### PR TITLE
Refactor trigger

### PR DESCRIPTION
**Pull request by yongpingchen**
- Refactor constructor of Trigger
- Move fromJson to JsonUtilities as jsonToTrigger
- Add small tests to jsonToTrigger method

### CODE REVIEW

File: src/internal/JsonUtilities.ts [hunk 1]
```diff
@@ -307,17 +307,25 @@ export function triggeredCommandToJson(cmd: TriggerCommandObject): Object {
...
-    trigger.disabled = obj.disabled === undefined ? null : obj.disabled;
-    trigger.disabledReason = obj.disabledReason ? obj.disabledReason : null;
-    trigger.title = obj.title ? obj.title : null;
-    trigger.description = obj.description ? obj.description : null;
-    trigger.metadata = obj.metadata ? obj.metadata : null;
-    return trigger;
+    if (!!obj.triggerID
+        && !!obj.predicate
+        && (obj.disabled != undefined && obj.disabled != null)) {
+        // TODO: need to move Predicate.fromJson
```
Comment (shinsuke-nara) [inline src/internal/JsonUtilities.ts]: Is this TODO comment required yet?

File: src/internal/JsonUtilities.ts [hunk 2]
```diff
@@ -307,17 +307,25 @@ export function triggeredCommandToJson(cmd: TriggerCommandObject): Object {
...
+        let predicate: Predicate = Predicate.fromJson(obj.predicate);
+        let trigger = new Trigger(obj.triggerID, predicate, obj.disabled);
+        if (!!obj.command) {
+            trigger.command = jsonToCommand(obj.command);
+        }
+        if (!!obj.serverCode) {
+            // TODO: need to move ServerCode.fromJson
```
Comment (shinsuke-nara) [inline src/internal/JsonUtilities.ts]: Is this TODO comment required yet?

File: src/ops/TriggerOps.ts [hunk 3]
```diff
@@ -88,9 +88,11 @@ export default class TriggerOps extends BaseOp {
             command.metadata = commandRequest.metadata;
 
             this.postTrigger(requestBody).then((res:Response)=>{
-                var trigger = new Trigger(requestObject.predicate, command, null);
-                trigger.triggerID = (<any>res).body.triggerID;
-                trigger.disabled = false;
+                var trigger = new Trigger(
+                    (<any>res).body.triggerID,
+                    requestObject.predicate,
```
Comment (shinsuke-nara) [inline src/ops/TriggerOps.ts]: In thing-if AndroidSDK and thing-if iOSSDK, `postTrigger()` method gets trigger from server with `getTrigger()` method. Why do not we get trigger with `getTrigger` in thing-if JSSDK?

File: src/internal/JsonUtilities.ts (hunk 1 above)
Comment (yongpingchen) [inline src/internal/JsonUtilities.ts]: Yes. Still need to move Predicate.fromJson to `JsonUtilities.ts`, it will be in another PR.

File: src/internal/JsonUtilities.ts (hunk 2 above)
Comment (yongpingchen) [inline src/internal/JsonUtilities.ts]: Yes. I will move toJson in another PR.

File: src/ops/TriggerOps.ts (hunk 3 above)
Comment (yongpingchen) [inline src/ops/TriggerOps.ts]: @shinsuke-nara. I will fixed the difference in another PR(fix trigger APIs)
